.github/scripts/transform_dbt_test_results.py

```diff
@@ -929,6 +954,104 @@ def create_list(
             for township_result in test_result.split_by_township()
         ]
 
+    def to_dict(self) -> typing.Dict:
+        return dataclasses.asdict(self)
+
+
+@dataclasses.dataclass
+class TestRunFailingRowMetadata:
+    """Metadata object storing information about row-level individual test
+    failures in a run."""
+
+    # Fields that identify the run
+    run_id: str
+    run_year: str  # Duplicated with TestRunMetadata for partitioning
+    # Fields that identify the test
+    test_name: str
+    table_name: str
+    column_name: typing.Optional[str]
+    category: str
+    description: str
+    # Fields that identify the failing row. Some of these can occasionally
+    # be arrays for tests that query multiple rows (e.g. uniqueness tests)
+    # so for consistency we set them to always be arrays, even when there
+    # is only one value
+    parid: typing.Optional[str]
+    taxyr: typing.Optional[str]
+    card: typing.Optional[typing.List[int]]
+    lline: typing.Optional[typing.List[int]]
+    class_: typing.Optional[typing.List[str]]
```

Comment:
I don't love that we need to compromise by making these columns arrays, since I find array columns to be annoying to deal with, but it strikes me as the best way to reconcile the two different approaches to these columns.